src/main/drivers/accgyro_mpu.c

```diff
@@ -234,8 +234,7 @@ static bool detectSPISensorsAndUpdateDetectionResult(gyroDev_t *gyro, const sens
 #else
     UNUSED(bus);
 #endif
-    if (mpu6000SpiDetect(&gyro->bus)) {
-        gyro->mpuDetectionResult.sensor = MPU_60x0_SPI;
+    if ((gyro->mpuDetectionResult.sensor = mpu6000SpiDetect(&gyro->bus))) {
```

Comment:
Call sequence is:
```
gyroInit
    mpuDetect()
        detectSPISensorsAndUpdateDetectionResult() which sets gyro->mpuDetectionResult.sensor if gyro detected
            mpu6500SpiDetect whichsets static mpuDetected if gyro detected
    gyroDetect
        mpu6500SpiGyroDetect  checks if gyro->mpuDetectionResult.sensor == mpuDetected
 ```

So I maintain your change is not required. However this investigation was useful since it revealed a `static` variable that I had overlooked - `mpuDetected`. This static variable will break multi-gyro support and also seems totally unnecessary - it seems like this was part of the lazy initialisation patter that plagued the inherited code.